app/src/main/kotlin/org/fossify/calendar/activities/EventActivity.kt

```diff
@@ -713,6 +714,27 @@ class EventActivity : SimpleActivity() {
             mEventEndDateTime = Formatter.getDateTimeFromTS(realStart + duration)
         }
 
+        // If editing an all-day event, set default times so when user disables all-day,
+        // the event uses proper default start time and duration while making sure to preserve the correct end date
+        if (mEvent.getIsAllDay()) {
+            val defaultStartTS = getNewEventTimestampFromCode(Formatter.getDayCodeFromDateTime(mEventStartDateTime))
+            val defaultStartTime = Formatter.getDateTimeFromTS(defaultStartTS)
+            val defaultDurationMinutes = config.defaultDuration
+            val endTime = defaultStartTime.plusMinutes(defaultDurationMinutes)
+            mEventStartDateTime = mEventStartDateTime.withTime(
+                defaultStartTime.hourOfDay,
+                defaultStartTime.minuteOfHour,
+                0,
+                0
+            )
+            mEventEndDateTime = mEventEndDateTime.withTime(
+                endTime.hourOfDay,
+                endTime.minuteOfHour,
+                0,
+                0
+            )
+        }
+
```

Comment:
It'll be safer to do this one-time migration directly in `toggledAllDay()`. Right now, this block will run unconditionally and might lead to subtle bugs later (or now if existing code assumes `mEventStartDateTime` and `mEventEndDateTime` reflect the DB values).